extensions/binding/org.eclipse.smarthome.binding.coapconnectedlighting/src/main/java/org/eclipse/smarthome/binding/coapconnectedlighting/handler/CoAPConnectedLightingHandler.java

```diff
@@ -0,0 +1,361 @@
+            updateStatus(ThingStatus.OFFLINE);
+            return;
+        }
+
+        stringChannelTemperatureUID = new ChannelUID(getThing().getUID(), CHANNEL_Temperature);
+        stringChannelHumidityUID = new ChannelUID(getThing().getUID(), CHANNEL_Humidity);
+        stringChannelPressureUID = new ChannelUID(getThing().getUID(), CHANNEL_Pressure);
+
+        refresh = new BigDecimal(60);
+
+        // Get the device properties
+        Map<String, String> deviceProperties = editProperties();
+
+        // Add the IP Address property
+        deviceProperties.put("IP Address", COAP_SERVER_IP_ADDRESS);
+
+        // Get the device functionality
+        coapClient.setURI("coap://" + COAP_SERVER_IP_ADDRESS + ":" + COAP_SERVER_PORT);
+        Set<WebLink> weblink = coapClient.discover();
+
+        if (weblink != null) {
+            Iterator<WebLink> links = weblink.iterator();
+
+            StringBuilder weblinkString = new StringBuilder();
+
+            while (links.hasNext()) {
+                weblinkString.append(":" + links.next().getURI());
+            }
+
+            String resourcesCombined = weblinkString.toString();
+            if (resourcesCombined.contains("actuators") && resourcesCombined.contains("sensors")) {
+                deviceProperties.put("Is a Sensor ?", "Yes");
+                deviceProperties.put("Is a Light ?", "Yes");
+            } else {
+                if (resourcesCombined.contains("actuators")) {
+                    deviceProperties.put("Is a Light ?", "Yes");
+                    deviceProperties.put("Is a Sensor ?", "No");
+                } else if (resourcesCombined.contains("sensors")) {
+                    deviceProperties.put("Is a Sensor ?", "Yes");
+                    deviceProperties.put("Is a Light ?", "No");
+                } else {
+                    deviceProperties.put("Is a Sensor ?", "No");
+                    deviceProperties.put("Is a Light ?", "No");
+                }
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Cannot identify the device functionality. Well-Known Core Discovery failed. Please remove the Thing and re-scan the device.");
+        }
+
+        // Update the device properties
+        updateProperties(deviceProperties);
+    }
+
+    private void getTemprature() {
+        coapClient.setURI(
+                "coap://" + COAP_SERVER_IP_ADDRESS + ":" + COAP_SERVER_PORT + "/InternetOfTiny/sensors/temperature");
+        logger.debug(coapClient.getURI());
+        CoapResponse clientResponseTemperature = coapClient.get(TEXT_PLAIN);
+
+        String temperatureText = clientResponseTemperature.getResponseText();
+        String temperatureFirstHalf = temperatureText.substring(0, temperatureText.lastIndexOf(','));
+        String temperature = temperatureFirstHalf.substring(temperatureFirstHalf.lastIndexOf(',') + 1);
```

Comment:
Line 157 to 161 are duplicated by blocks 173 - 176 and 188 - 191 and may be refactored to a private method which will retrieve the CoapResponse and parse and return a `new DecimalType` (which imho should also be part of the method).
Even the #setUri call and the logger should be part of this method. like:

```
private DecimalType getDecimalValueFromSensor(String sensorType) {
    coapClient.setURI(
        "coap://" + COAP_SERVER_IP_ADDRESS + ":" + COAP_SERVER_PORT + "/InternetOfTiny/sensors/" + sensorType);
    logger.debug(coapClient.getURI());
    CoapResponse clientResponse = coapClient.get(TEXT_PLAIN);

    String valueText = clientResponse.getResponseText();
    String valueFirstHalf = valueText.substring(0, valueText.lastIndexOf(','));
    String value = valueFirstHalf.substring(valueFirstHalf.lastIndexOf(',') + 1);

    return new DecimalType(value);
}

```